### PR TITLE
[k8sconfiguration] Parameter Validation and Table Formatting

### PR DESCRIPTION
**Pull request by jonathan-innis**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
k8sconfiguration

**Comment by yungezz** [comment]
hi @qwordy could you help review this PR? thanks

File: src/k8sconfiguration/azext_k8sconfiguration/_format.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    for result in results:
+        formatted_result.append(
+            OrderedDict([
+                ('name', result['name']),
+                ('repositoryUrl', result['repositoryUrl']),
+                ('operatorName', result['operatorInstanceName']),
+                ('operatorNamespace', result['operatorNamespace']),
+                ('scope', result['operatorScope']),
+                ('provisioningState', result['provisioningState'])
+            ])
```
Comment (qwordy) [inline src/k8sconfiguration/azext_k8sconfiguration/_format.py:20]: Duplicate code. Maybe you can add a function.

File: src/k8sconfiguration/azext_k8sconfiguration/_format.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+def k8sconfiguration_show_table_format(result):
+    return OrderedDict([
+        ('name', result['name']),
+        ('repositoryUrl', result['repositoryUrl']),
+        ('operatorName', result['operatorInstanceName']),
+        ('operatorNamespace', result['operatorNamespace']),
+        ('scope', result['operatorScope']),
+        ('provisioningState', result['provisioningState'])
+    ])
```
Comment (qwordy) [inline src/k8sconfiguration/azext_k8sconfiguration/_format.py:33]: Duplicate code. Maybe you can add a function.

File: src/k8sconfiguration/azext_k8sconfiguration/_params.py [hunk 3]
```diff
@@ -42,7 +51,7 @@ def load_arguments(self, _):
         c.argument('operator_params',
                    help='Parameters for the Operator')
         c.argument('ssh_private_key',
-                   help='Specify private ssh key for private repository sync (either base64 encoded or raw)')
+                   help='Specify base64-encoded private ssh key for private repository sync')
```
Comment (qwordy) [inline src/k8sconfiguration/azext_k8sconfiguration/_params.py:54]:
```suggestion
                   help='Specify Base64-encoded private ssh key for private repository sync')
```

**Review by qwordy — APPROVED** [review]

File: src/k8sconfiguration/azext_k8sconfiguration/_params.py (hunk 3 above)
Comment (jonathan-innis) [reply src/k8sconfiguration/azext_k8sconfiguration/_params.py]: Updated based on comment

**Comment by qwordy** [comment]
@jonathan-innis Thank you for resolving the comments.

**Pull request merged**